### PR TITLE
Diegoro1/feature/user location permission loader

### PR DESCRIPTION
**Pull request by diegoro1**
Small changes to the front end of the application:
* Added a loader/spinner with an overlay to show the user the map is loading/waiting for their location.
* Added some extra metadata to index.html
* Slightly refactored maps.js  
* Separated the CSS from the HTML

Screenshot of changes attached 
Screen Shot 2022-02-25 at 1 41 36 AM

Let me know if this is the direction you're taking with the web app! I built this branch off of the "remove commented out code" branch (as you can see 🙂).

### CODE REVIEW

File: priv/static/map.js [hunk 1]
```diff
@@ -14,7 +14,37 @@ fetch("tournaments").then((resp) => {
...
+
 // Set position based on user location, if they grant it.
-navigator.geolocation.getCurrentPosition(resp => {
+// Removes overlay and loader during success or failure.
+function getLocation() {
+    if (navigator.geolocation) {
+        navigator.geolocation.getCurrentPosition(setPositition, showError)
+    }
+    else {
+        alert("Please unable the browser to know your location or Geolocation is not supported by this browser.");
```
Comment (perrycate) [inline priv/static/map.js:34]: typo: enable/unable

**Comment by perrycate** [comment]
Seems like if the permission is denied from chromium or chrome (at least on my linux box) the tab just hangs?

**Comment by diegoro1** [comment]
> Seems like if the permission is denied from chromium or chrome (at least on my linux box) the tab just hangs?

What happens if you accept the permission in chromium? I can't seem to replicate the problem in any browser on my machine 🤔.

**Comment by perrycate** [comment]
Whelp I couldn't replicate the issue today so...

**Pull request merged**